### PR TITLE
Add support for comments in hypervisor.args

### LINKED ISSUE #35: Support comments in hypervisor.args
Support lines starting with `#` as comments.

### PR DESCRIPTION
**Pull request by devimc**
When a word begins with '#' that word and all remaining characters
will be treated as comments (will be ignored)

example:
input text:
```
line #comment
#comment
line#
line#text
line\#text
line #
line # comment
line \# #comment
```
Result:
```
line

line#
line#text
line\#text
line
line
line \#
```

Fixes #35

Signed-off-by: Julio Montes <julio.montes@intel.com>

### CODE REVIEW

File: src/hypervisor.c [hunk 1]
```diff
@@ -160,6 +160,23 @@ cc_oci_expand_cmdline (struct cc_oci_config *config,
 			}
 		}
 
+		/* first character is '#', line is empty */
```
Comment (grahamwhaley) [inline src/hypervisor.c]: Technically, the line is not 'empty' is it - it is 'a comment' ?

**Comment by grahamwhaley** [comment]
Hi.
 Thanks for doing this - and nicely done with the tests cases!
 I scanned the code and found it hard to figure out if comments extended to the end of the line (seeing a while loop in there threw me for instance) and what the exact rules were around when a '#' is seen to be the start of a comment. Could you maybe add some syntax or examples to the commit message?

File: src/hypervisor.c (hunk 1 above)
Comment (devimc) [inline src/hypervisor.c]: hehehe yes, sorry my bad , line is a comment

**Comment by sameo** [comment]
Looks good to me.

**Comment by sameo** [comment]
@grahamwhaley Does this look ok to you ?

**Comment by jcvenegas** [comment]
+1

**Comment by gorozco1** [comment]
+1

**Comment by sameo** [comment]
@devimc Please rebase this patch and we'll merge it.

**Pull request merged**